File: packages/java/src/main/java/com/peruspatial/ubigeo/GetDepartment.java
package com.peruspatial.ubigeo;

import com.peruspatial.data.UbigeoData;

public final class GetDepartment {
  private GetDepartment() {}

  public static UbigeoRecord getDepartment(String code) {
    if (code == null) return null;
    for (UbigeoRecord r : UbigeoData.DEPARTMENTS) {
      if (r.code().equals(code)) return r;
    }
    return null;
  }
}

File: packages/rust/src/ubigeo/get_department.rs
use super::record::{find, UbigeoRecord};
use crate::data::ubigeo::DEPARTMENTS;

pub fn get_department(code: &str) -> Option<UbigeoRecord> {
    find(DEPARTMENTS, code)
}

pub fn get_region(code: &str) -> Option<UbigeoRecord> {
    get_department(code)
}
